src/lib_ccx/matroska.c

```diff
@@ -738,14 +738,35 @@ void parse_segment(struct matroska_ctx* mkv_ctx)
     }
 }
 
+char *filename_without_ext(char* filename) {
+    char *returnname;
+    char *lastdot;
+    if (filename == NULL)
+         return filename;
+    if ((returnname = malloc (strlen (filename) + 1)) == NULL)
+        return filename;
+    strcpy (returnname, filename);
+    lastdot = strrchr (returnname, '.');
+    if (lastdot != NULL)
+        *lastdot = '\0';
+    return returnname;
+}
+
 char* generate_filename_from_track(struct matroska_ctx* mkv_ctx, struct matroska_sub_track* track)
-{
-    char* buf = malloc(sizeof(char) * 200);
+{ 
+  char* filename = malloc(sizeof(char)*(MAX_FILE_NAME_SIZE-30));
```

Comment:
Actually I kept maximum size to 200, this 30 is for any additional format addition like eng_srt or any other language format. Ths means effectively 170 characters from original file name and 30 chars to define our srt extension type
Originally it was 200 chars so I kept it to 200 chars. Do you want any change.